src/app/charts/chart-base/chart-base.component.ts

```diff
@@ -191,26 +207,53 @@ export class ChartBaseComponent implements AfterViewInit, OnDestroy {
     this.brush = d3
       .brushX()
       .extent([[0, 0], [this.width, this.height]])
-      .on('end', () => this.brushed(this.xScale))
+      .on('end', () => this.brushed())
+
+    if (this.sensorDataTimeFrame) {
+      this.updateBrushFromSensorDates()
+    }
 
     this.chart
       .append('g')
       .attr('class', 'brush')
       .call(this.brush)
+      .transition()
+      .duration(1000)
       .call(
         this.brush.move,
-        this.brushExtent
-          ? this.brushExtent
-          : [this.width - this.brushWidthDefault, this.width]
+        !this.brushExtentFromSensors
+          ? [this.width - this.brushWidthDefault, this.width]
+          : this.brushExtentFromSensors
       )
   }
 
-  private brushed(xScale) {
-    this.brushExtent = d3.event.selection
+  private brushed() {
+    if (!this.brushUpdatedFromSensors) {
+      this.brushExtent = d3.event.selection
+
```

Comment:
`this.brushUpdatedFromSensors` doesn't seem to set to `true` after `brushed` is executed. Is this intentional?

Reply:
The `this.brushUpdatedFromSensors` is set to true only when brush is updated from sensor data dates, when sensor data dates are updated or when the chart is redrawn. This is not updated the user moves the brush. This is to maintain the brush when the timeframe is changed or window size is changed.